.github/workflows/test-code.yml

```diff
@@ -38,3 +38,10 @@ jobs:
       - name: Test solution (debug config)
         working-directory: src
         run: dotnet test --no-restore
+  buildcheck:
+    needs:
+      - test
+    runs-on: ubuntu-latest
+    steps:
+      - name: Pass build check
```

Comment:
That is, is it a edge case when all the work has already been done in the dependent step `test`? Also why do you use `runs-on: ubuntu-latest` ? Does this narrow the check?

Reply:
Ah, buildcheck is just a name for ordinary action.

Reply:
What this job does is wait for the prior job (which functions as two separate jobs) to fully complete, and if both pass, then pass a "buildcheck" status back to the PR.  Without this, the two separate jobs report status checks separately for each OS.

`runs-on` is a required field for a job, as is `steps`.  As you can see, this job doesn't actually "do" anything.  It immediately passes ... so long as the prior steps both pass.

Reply:
Just wondering. Is it essential that the name of the job be unique among all actions? Now the same name `buildcheck` is used inside two actions, but the actions themselves never fire together because they are responsible for different parts of the project. 

Could there be two entries with the same name?
![изображение](https://user-images.githubusercontent.com/21261007/97023671-80fb6400-155e-11eb-99c9-be56777cac42.png)

Reply:
And one more question. Don't get me wrong, I'm just learning from your examples. Until now, any action could exit with a code of 1 (error) and would it still allow us to merge? Only after the buildcheck is turned on, a strict check could be introduced to forbid merging?

Reply:
> Without this, the two separate jobs report status checks separately for each OS.

That is, did I understand correctly that the successful status will be set on PR only when the tests pass successfully on all operating systems? If that's true, then I'm just a little confused by the `runs-on: ubuntu-latest` line. This gives the impression that this will only check the success status on one OS.

Reply:
> Just wondering. Is it essential that the name of the job be unique among all actions? Now the same name buildcheck is used inside two actions, but the actions themselves never fire together because they are responsible for different parts of the project.

Well, I tested both workflows and it didn't add two "buildcheck" items in the list.  So it would seem to match based on name.

I think the safest way to go would to be to merge the two test workflow files together.  But, we cannot have path filtering on different jobs in the same workflow.  So, it would always run both code and documentation checks.  (Which considering that workflow runtime is free for public repos is not a big deal.)

I do think this PR will work as-is, but we may find otherwise.  For instance, what happens if both code and docs are changed in the same PR?  Does it fail if any fail?  Does it fail only if the last to finish fails?  I don't know for sure.  I'm just hoping it works like we want it to.

> Until now, any action could exit with a code of 1 (error) and would it still allow us to merge?

Not exactly.  Right now it is requiring "continuous-integration/appveyor/pr" to pass (unless you just changed it).  But as an 'administrator' on this repo, you and I can merge regardless of status checks.

> That is, did I understand correctly that the successful status will be set on PR only when the tests pass successfully on all operating systems? If that's true, then I'm just a little confused by the runs-on: ubuntu-latest line. This gives the impression that this will only check the success status on one OS.

That line with the `runs-on` indicates what platform that that "buildcheck" job runs on.  The "buildcheck" job requires the "test" job to fully pass (including both matrix steps) before it will run at all.  So if windows or ubuntu fails in the first job, then the "buildcheck" job will never run, regardless of what platform that the "buildcheck" job is programmed to run on.

Reply:
>  For instance, what happens if both code and docs are changed in the same PR? Does it fail if any fail? Does it fail only if the last to finish fails?

Interesting question. I don't know too.

> Right now it is requiring "continuous-integration/appveyor/pr" to pass (unless you just changed it)

I khow abou it. I meant a situation where we already turn it off. It's just that before that I thought that any failed action prevents merge. I didn't know about picking special action (or actions) as a build status indicator.

>  The "buildcheck" job requires the "test" job to fully pass (including both matrix steps) before it will run at all. So if windows or ubuntu fails in the first job, then the "buildcheck" job will never run, regardless of what platform that the "buildcheck" job is programmed to run on.

OK. Exactly what is needed.